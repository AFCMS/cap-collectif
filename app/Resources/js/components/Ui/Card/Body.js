// @flow
import styled from 'styled-components';

const Body = styled.div.attrs({
  className: 'card-body',
})`
  display: flex;
  padding: 15px;
  flex: 1 0 auto;
  flex-direction: column;

  hr {
    margin: 15px 0 10px;
  }

  @media print {
    display: block;
    padding: 0;
  }
`;

export default Body;
