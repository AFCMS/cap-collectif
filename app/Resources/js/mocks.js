// @flow
/* eslint-env jest */
import { type FormProps } from 'redux-form';
import { type IntlShape } from 'react-intl';
import type { RelayPaginationProp, RelayRefetchProp } from 'react-relay';

export const $refType: any = null;
export const $fragmentRefs: any = null;

export const intlMock: IntlShape = {
  locale: 'fr-FR',
  formats: {},
  messages: {},
  now: () => 0,
  formatHTMLMessage: message => String(message),
  formatPlural: message => String(message),
  formatNumber: message => String(message),
  formatRelative: message => String(message),
  formatTime: message => String(message),
  formatDate: message => String(message),
  formatMessage: message => String(message.id),
};

export const formMock: FormProps = {
  anyTouched: false,
  array: {
    insert: jest.fn(),
    move: jest.fn(),
    pop: jest.fn(),
    push: jest.fn(),
    remove: jest.fn(),
    removeAll: jest.fn(),
    shift: jest.fn(),
    splice: jest.fn(),
    swap: jest.fn(),
    unshift: jest.fn(),
  },
  asyncValidate: jest.fn(),
  asyncValidating: false,
  autofill: jest.fn(),
  blur: jest.fn(),
  change: jest.fn(),
  clearAsyncError: jest.fn(),
  clearSubmit: jest.fn(),
  destroy: jest.fn(),
  dirty: false,
  dispatch: jest.fn(),
  error: null,
  form: 'formName',
  handleSubmit: jest.fn(),
  initialize: jest.fn(),
  initialized: true,
  initialValues: {},
  invalid: false,
  pristine: true,
  reset: jest.fn(),
  submitting: false,
  submitFailed: false,
  submitSucceeded: false,
  touch: jest.fn(),
  untouch: jest.fn(),
  valid: true,
  warning: null,
};

export const relayPaginationMock: RelayPaginationProp = {
  environment: {
    applyMutation: jest.fn(),
    sendMutation: jest.fn(),
    lookup: jest.fn(),
    sendQuery: jest.fn(),
    subscribe: jest.fn(),
    streamQuery: jest.fn(),
    retain: jest.fn(),
    unstable_internal: {},
  },
  hasMore: () => false,
  isLoading: () => false,
  loadMore: jest.fn(),
  refetchConnection: jest.fn(),
};

export const relayRefetchMock: RelayRefetchProp = {
  environment: {
    applyMutation: jest.fn(),
    sendMutation: jest.fn(),
    lookup: jest.fn(),
    sendQuery: jest.fn(),
    subscribe: jest.fn(),
    streamQuery: jest.fn(),
    retain: jest.fn(),
    unstable_internal: {},
  },
  refetch: jest.fn(),
};
