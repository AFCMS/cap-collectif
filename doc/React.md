### How to write localizable components

Just import the appropriate [component](https://github.com/yahoo/react-intl/wiki#the-react-intl-module) from `react-intl`

- For localizable text use
[`<FormattedMessage>`](https://github.com/yahoo/react-intl/wiki/Components#formattedmessage).

- For date and time:
[`<FormattedDate>`](https://github.com/yahoo/react-intl/wiki/Components#formatteddate)
[`<FormattedTime>`](https://github.com/yahoo/react-intl/wiki/Components#formattedtime)
[`<FormattedRelative>`](https://github.com/yahoo/react-intl/wiki/Components#formattedrelative)

- For numbers and currencies:
[`<FormattedNumber>`](https://github.com/yahoo/react-intl/wiki/Components#formattednumber)
[`<FormattedPlural>`](https://github.com/yahoo/react-intl/wiki/Components#formattedplural)

- If possible, do not use `<FormattedHTMLMessage>`, see how to use *Rich Text Formatting* with
[`<FormattedMessage>`](https://github.com/yahoo/react-intl/wiki/Components#formattedmessage)

- When you need an imperative formatting API, use the [`injectIntl`](https://github.com/yahoo/react-intl/wiki/API#injectintl) High-Order Component.
