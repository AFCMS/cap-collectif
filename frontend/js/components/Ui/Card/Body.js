// @flow
import styled, { css, type StyledComponent } from 'styled-components';

const Body: StyledComponent<
  { isHorizontal?: boolean, position?: 'relative' | 'absolute' },
  {},
  HTMLDivElement,
> = styled.div.attrs({
  className: 'card__body',
})`
  display: flex;
  padding: 15px;
  flex: 1 0 auto;
  position: ${props => (props.position ? props.position : '')};
  flex-direction: ${props => (props.isHorizontal ? 'row' : 'column')};
  word-wrap: break-word;
  ${({ isHorizontal }) =>
    isHorizontal &&
    css`
      align-items: center;
      justify-content: space-between;
    `}

  hr {
    margin: 15px 0 10px;
  }

  @media print {
    display: block;
    padding: 0;
  }
`;

export default Body;
