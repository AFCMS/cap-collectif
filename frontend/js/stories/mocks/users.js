// @flow

export const author = {
  id: 'user1',
  username: 'admin',
  media: {
    url: 'https://source.unsplash.com/random/150x150',
  },
  _links: {},
  vip: false,
  isViewer: false,
  isAdmin: true,
  isEvaluer: true,
};

export const followers = [
  {
    displayName: 'All Fred',
    isEmailConfirmed: true,
    media: {
      url: 'https://source.unsplash.com/collection/181462',
    },
    _links: {
      profile: 'https://capco.dev',
    },
    contributionsCount: 4,
  },
  {
    displayName: 'guemas',
    isEmailConfirmed: true,
    media: {
      url: 'https://source.unsplash.com/collection/181462',
    },
    _links: {
      profile: 'https://capco.dev',
    },
    contributionsCount: 0,
  },
  {
    displayName: 'Shezeboss',
    isEmailConfirmed: true,
    media: {
      url: 'https://source.unsplash.com/collection/181462',
    },
    _links: {
      profile: 'https://capco.dev',
    },
    contributionsCount: 4,
  },
  {
    displayName: 'AS',
    isEmailConfirmed: true,
    media: {
      url: 'https://source.unsplash.com/collection/181462',
    },
    _links: {
      profile: 'https://capco.dev',
    },
    contributionsCount: 240,
  },
  {
    displayName: 'Louisette',
    isEmailConfirmed: false,
    media: {
      url: 'https://source.unsplash.com/collection/181462',
    },
    _links: {
      profile: 'https://capco.dev',
    },
    contributionsCount: 1,
  },
  {
    contributionsCount: 1,
    isEmailConfirmed: true,
    _links: null,
    media: null,
  },
  {
    displayName: 'guemas',
    isEmailConfirmed: true,
    media: {
      url: 'https://source.unsplash.com/collection/181462',
    },
    _links: {
      profile: 'https://capco.dev',
    },
    contributionsCount: 0,
  },
  {
    displayName: 'Shezeboss',
    isEmailConfirmed: true,
    media: {
      url: 'https://source.unsplash.com/collection/181462',
    },
    _links: {
      profile: 'https://capco.dev',
    },
    contributionsCount: 4,
  },
  {
    displayName: 'AS',
    isEmailConfirmed: true,
    media: {
      url: 'https://source.unsplash.com/collection/181462',
    },
    _links: {
      profile: 'https://capco.dev',
    },
    contributionsCount: 240,
  },
];
