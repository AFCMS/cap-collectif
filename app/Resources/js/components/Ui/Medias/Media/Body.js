// @flow
import React, { PureComponent } from 'react';
import { Media as MediaBtsp } from 'react-bootstrap';

type Props = {
  children?: any,
};

export default class Body extends PureComponent<Props> {
  render() {
    const { children } = this.props;

    return <MediaBtsp.Body>{children}</MediaBtsp.Body>;
  }
}
