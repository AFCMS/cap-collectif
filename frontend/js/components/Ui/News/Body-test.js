// @flow
/* eslint-env jest */
import React from 'react';
import { shallow } from 'enzyme';
import { Body } from './Body';
import { news, withoutParaph } from '../../../stories/mocks/news';

describe('Body />', () => {
  it('should render correctly', () => {
    const wrapper = shallow(<Body news={news} />);
    expect(wrapper).toMatchSnapshot();
  });
  it('should render correctly but without paraph in body', () => {
    const wrapper = shallow(<Body news={withoutParaph} />);
    expect(wrapper).toMatchSnapshot();
  });
});
