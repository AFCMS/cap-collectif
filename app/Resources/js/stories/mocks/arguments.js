// @flow
import { author } from './users';

export const argument = {
  user: author,
  trashedStatus: '',
  createdAt: '1 mars 2016 à 00:00',
  publishedAt: '1 mars 2016 à 00:00',
  body:
    'In placeat reiciendis ut. Officiis praesentium quia minima ut tenetur officiis. Eaque fugit voluptates temporibus suscipit provident culpa culpa. Magni recusandae dolorem aut id.',
  votes: { totalCount: 50 },
};

export const argumentsData = [
  { ...argument, user: { ...argument.user, vip: true } },
  { ...argument },
  { ...argument, trashedStatus: 'INVISIBLE' },
  { ...argument },
];
