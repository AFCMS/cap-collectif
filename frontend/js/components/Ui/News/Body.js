// @flow
import React from 'react';
import styled, { type StyledComponent } from 'styled-components';
import { FormattedMessage } from 'react-intl';
import CroppedLabel from '../Labels/CroppedLabel';
import { type NewsProps } from './News';

type Props = {|
  news: NewsProps,
|};

const BodyContainer: StyledComponent<{}, {}, HTMLDivElement> = styled.div`
  display: flex;
  flex-direction: column;
  flex: 2;
  width: 100%;
  height: 100%;
  margin: 10px;

  @media (max-width: 768px) {
    font-size: 25px;
    margin-left: 25px;

    .news--body-text_container {
      width: 100%;
      height: 100%;
      display: flex;
    }
  }

  @media (max-width: 992px) {
    .news--body-text_container {
      width: 100%;
      height: 100%;
      margin: 10px;
    }
  }

  @media (min-width: 1200px) {
    .news--subtitle {
      line-height: 60px;
    }
  }
`;

const Title: StyledComponent<{}, {}, HTMLHeadingElement> = styled.h1`
  font-size: 30px;
  justify-content: flex-start;
  padding: 0;
  margin: 0;
  display: flex;

  @media (max-width: 992px) {
    .news--title {
      line-height: 25px;
      font-size: 25px;
    }
  }
`;

const NewsInfos: StyledComponent<{}, {}, HTMLDivElement> = styled.div`
  display: block;
  color: gray;
  font-size: 14px;
  line-height: 30px;
  .label {
    background-color: gray;
    border-radius: 4px;
    margin: 5px;
    font-size: 12px;
    text-shadow: 1px 1px 2px black;
    cursor: pointer;
  }
`;

const Paraph: StyledComponent<{}, {}, HTMLDivElement> = styled.div`
  font-size: 20px;
  margin-left: 5px;
`;

export function Body({ news }: Props) {
  return (
    <React.Fragment>
      <BodyContainer>
        <Title>
          <a href={news.url}>{news.title}</a>
        </Title>

        <NewsInfos>
          <CroppedLabel label={{ name: news.theme, color: '#66666e' }} />
          <FormattedMessage id="global.byDate" values={{ date: news.date.released }} />{' '}
          <FormattedMessage id="global.byAuthor" values={{ author: news.user.username }} />
        </NewsInfos>
        <Paraph>{news.body}</Paraph>
      </BodyContainer>
    </React.Fragment>
  );
}

export default Body;
