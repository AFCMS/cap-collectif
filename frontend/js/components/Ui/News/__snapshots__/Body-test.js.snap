// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Body /> should render correctly 1`] = `
<Fragment>
  <Body__BodyContainer>
    <Body__Title>
      <a
        href="#"
      >
        Je suis le titre
      </a>
    </Body__Title>
    <Body__NewsInfos>
      <CroppedLabel
        label={
          Object {
            "color": "success",
            "name": "Test & Democratie",
          }
        }
        maxSize={9}
      />
      <FormattedMessage
        id="global.byDate"
        values={
          Object {
            "date": "2019-11-09",
          }
        }
      />
       
      <FormattedMessage
        id="global.byAuthor"
        values={
          Object {
            "author": "admin",
          }
        }
      />
    </Body__NewsInfos>
    <Body__Paraph>
       Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.
    </Body__Paraph>
  </Body__BodyContainer>
</Fragment>
`;

exports[`Body /> should render correctly but without paraph in body 1`] = `
<Fragment>
  <Body__BodyContainer>
    <Body__Title>
      <a
        href="#"
      >
        un titre pour le deuxi√®me cas
      </a>
    </Body__Title>
    <Body__NewsInfos>
      <CroppedLabel
        label={
          Object {
            "color": "success",
            "name": "Test et Jest",
          }
        }
        maxSize={9}
      />
      <FormattedMessage
        id="global.byDate"
        values={
          Object {
            "date": "19/12/2019",
          }
        }
      />
       
      <FormattedMessage
        id="global.byAuthor"
        values={
          Object {
            "author": "admin",
          }
        }
      />
    </Body__NewsInfos>
    <Body__Paraph />
  </Body__BodyContainer>
</Fragment>
`;
